src/routes/database.ts

```diff
@@ -1,320 +1,430 @@
+async function userFeedback(
+  request: UserFeedbackRequest,
+  h: ResponseToolkit,
+  d: any
+): Promise<ResponseObject> {
+  let validation = new Validation();
+  await validation.validateUserFeedback(request.payload);
+
+  if (validation.fail()) {
+    return h.response(validation.errors).code(400);
+  }
+
+  let payload: any = request.payload;
+
+  try {
+    let userFeedback = await createUserFeedback({
+      question_1: payload.question1,
+      question_2: payload.question2,
+      question_3: payload.question3,
+      question_4: payload.question4,
+      user_id: request.auth.credentials.user_id,
+    });
+
+    return h.response(userFeedback).code(200);
+  } catch (err: any) {
+    console.log(err.message);
+    return h.response("internal server error!").code(500);
+  }
 }
 
 export const databaseRoutes: ServerRoute[] = [
-    /** Public APIs */
-    // Register a new account
-    { method: "POST", path: "/api/user/register", handler: registerUser, options: { auth: false } },
-    // Request a password reset for an email address
-    { method: "POST", path: "/api/user/password-reset", handler: resetPassword, options: { auth: false } },
-    // Login user and retrieve a token
-    { method: "POST", path: "/api/token", handler: loginUser, options: { auth: false } },
-
-    /** Authenticated users only */
-    // Return logged in user's details
-    { method: "GET", path: "/api/user/details", handler: getAuthUserDetails },
-    // Allow user to change their email address
-    { method: "POST", path: "/api/user/email", handler: changeEmail },
-    // Allow user to change their details
-    { method: "POST", path: "/api/user/details", handler: changeUserDetail },
-    // Allow logged in user to change their password
-    { method: "POST", path: "/api/user/password", handler: changePassword },
-
+  /** Public APIs */
+  // Register a new account
+  {
+    method: "POST",
+    path: "/api/user/register",
+    handler: registerUser,
+    options: { auth: false },
+  },
+  // Request a password reset for an email address
+  {
+    method: "POST",
+    path: "/api/user/password-reset",
+    handler: resetPassword,
+    options: { auth: false },
+  },
+  // Login user and retrieve a token
+  {
+    method: "POST",
+    path: "/api/token",
+    handler: loginUser,
+    options: { auth: false },
+  },
+
+  /** Authenticated users only */
+  // Return logged in user's details
+  { method: "GET", path: "/api/user/details", handler: getAuthUserDetails },
+  // Allow user to change their email address
+  { method: "POST", path: "/api/user/email", handler: changeEmail },
+  // Allow user to change their details
+  { method: "POST", path: "/api/user/details", handler: changeUserDetail },
+  // Allow logged in user to change their password
+  { method: "POST", path: "/api/user/password", handler: changePassword },
+  // Allow logged in user to submit feedback
+  { method: "POST", path: "/api/user/feedback", handler: userFeedback },
```

Comment:
This method doesn't actually require the user to be authenticated currently. Is this what we want? (from our current use case, I think we probably only want the API available to logged in users, but I'm not sure about this)

So we should make a decision about this and then either move this route up to the 'public APIs' section, or add an authentication check to the handler so that only authenticated users can submit feedback.

Reply:
@ms0ur1s Ah yep, you're right that it will require authentication. I would also add the auth.credentials.user_id field to the UserFeedbackRequest type though, so that it's properly type checked before the handler runs (e.g. see UserDetailsRequest type)

Reply:
Ah, nice one @rogup, missed that that.